go-controller/pkg/ovn/egressip.go

```diff
@@ -376,14 +376,25 @@ func (oc *Controller) reconcileEgressIPPod(old, new *v1.Pod) (err error) {
 			// If the podSelector is not empty then check if the pod stopped
 			// matching. If the pod was deleted, "new" will be nil and
 			// newPodLabels will not match, so this is should cover that case.
-			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) && podSelector.Matches(oldPodLabels) {
-				return oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+			if !podSelector.Empty() && !podSelector.Matches(newPodLabels) {
+				if podSelector.Matches(oldPodLabels) {
+					err = oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+					if err != nil {
+						return err
+					}
+				}
+				// egressIP has a podSelector that pod does not match: skip it
+				continue
 			}
 			// If the podSelector is empty (i.e: the EgressIP object is intended
 			// to match all pods in the namespace) and the pod has been deleted:
 			// "new" will be nil and we need to remove the setup
 			if podSelector.Empty() && new == nil {
-				return oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+				err = oc.deletePodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, oldPod)
+				if err != nil {
+					return err
+				}
+				continue
 			}
 			// For all else, perform a setup for the pod
 			return oc.addPodEgressIPAssignments(egressIP.Name, egressIP.Status.Items, newPod)
```

Comment:
Come to think of it we will probably need a continue statement here too. We need to cover the following case:

1. Egress IP 1 exists with a pod selector which matches pod A 
2. Egress IP 2 is created with only a namespace selector matching pod A 
3. pod A is update and has its labels removed, which means: it should stop matching Egress IP 1 and start matching Egress IP 2. 

If we process Egress IP 2 before Egress IP 1 when the pod update happens, we need to continue here rather than return as to ensure that we remove the setup made for the pod w.r.t Egress IP 1.

Reply:
Ack. Since you already fixed this in https://github.com/ovn-org/ovn-kubernetes/pull/2734 let's abandon this PR.